server/src/test/java/org/elasticsearch/common/util/CollectionUtilsTests.java

```diff
@@ -78,6 +78,7 @@ public void testUniquify() {
         assertUniquify(List.of(1, 1, 1), Comparator.naturalOrder(), 1);
         assertUniquify(List.of(1, 2, 2, 3), Comparator.naturalOrder(), 3);
         assertUniquify(List.of(1, 2, 2, 2), Comparator.naturalOrder(), 2);
+        assertUniquify(List.of(1, 2, 2, 3, 3, 5), Comparator.naturalOrder(), 4);
```

Comment:
I think this needs a stronger test. I was thinking about this:
1. Generate random int `s` for the number of unique items in list
2. Generate `s` unique numbers 
3. For each random generate random number of repeats (1  to 10)
4. Put everything in the list
5. uniquify the list
6. compare resulting list with a list of only unique numbers that we have from 2

Reply:
I added a random test, see [62cdd9a](https://github.com/elastic/elasticsearch/pull/126889/commits/62cdd9abfc06eac63550691fe6d79a3cf3842bd0)